src/server/listener.cc

```diff
@@ -174,6 +174,7 @@ Listener::bind(const Address& address) {
         TRY(::listen(fd, backlog_));
         break;
     }
+    freeaddrinfo(addrs);
```

Comment:
This works in the common case, but will still fail to clean up resources if the "::listen()" fails 2 lines above (the 'TRY' macro will throw a different exception).

I suggest that you encapsulate 'getaddrinfo' into a RAII class so that it will clean itself up no matter how the method exits.